content/docs/2.3/scalers/azure-event-hub.md

```diff
@@ -30,6 +30,11 @@ triggers:
 - `consumerGroup` - Consumer group of event hub consumer. (default: `$default`)
 - `unprocessedEventThreshold` - : Average target value to trigger scaling actions. (default: 64)
 - `blobContainer` - Container name to store checkpoint. This is needed when a using an Event Hub application written in dotnet or java, and not an Azure function
+- `checkpointStrategy` - configure the checkpoint behaviour of different eventhub sdk's. (default: `""`)
+    - `""`: the default behaviour works for C#, older Python and Java eventhub implementations if a `blobContainer` is configured.
```

Comment:
We might want to rename this to `Default` or `NotSpecified`?

Reply:
you mean simply instead of an empty string, "NotSpecified" as parameter description, or also an explicit naming of the parameter in code? I wouldn't be so happy with "default" as the parameter name in the code. I think more and more libs will implement the metadata approach. Currently this is the default setting, due to backward compatibility

Reply:
I think I get it now, wit this line you mean that it will use this when it's not specified, is that correct?

So there's two things:
- `blobContainer` that is used for checkpointing and works for C#, older Python and Java eventhub implementations
- If nothing is specified, then `blobContainer` is used

Is that correct? If so, I would document `blobContainer` and add `(default: blobContainer)` on line 33 where we explain what `checkpointStrategy` is. Makes sense?

Reply:
The blobContainer description is currently not 100 percent correct. A `blobContainer` name is for all checkpointStrategies required except for azure Functions. The checkpointing for Azure Function is the only implementation which sets the containername automatically and cannot be overridden. So the old scaler logic was simply:

if blobContainer == "" then
 --> azureFunction
 else
  --> everything else (which works for c# and older frameworks)
end

So for backward compatibility, when no strategy is specified, then the "old checkpointing" (else path above) is taken.
For the same reason, if no blobContainer name is set, the azure Function checkpointing take place. Alternatively is it now possible to set the "azureFunction" strategy explicit. 

The new strategies requires a value for `checkpointStrategy`, no backward compatibility is necessary.

I can try to write a more clear description in the evening. The other possibility is, we require with keda 2.3 an explicit value in checkpointStrategy, and add a breaking Change comment. Then we need also an explicit name for the "old checkpointing" behavior. Maybe C#, (but this is also true for older implementations of other languages).

FYI: Yesterday someone mentioned  on the code PR that the new blobMetadata strategy works also for the JS implementation. We should add it in the list above, too.

Reply:
Let's use the approach with non-breaking changes but maybe provide some more information in the docs then indeed.

Thanks and sorry for dragging on but it's a pretty "big" change.